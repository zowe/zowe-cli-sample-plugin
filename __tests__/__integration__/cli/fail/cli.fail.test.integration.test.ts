/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import {runCliScript} from "../../../__src__/TestUtils";
import {ITestEnvironment} from "../../../__src__/environment/doc/response/ITestEnvironment";
import {TestEnvironment} from "../../../__src__/environment/TestEnvironment";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment;

describe("zowe-cli-sample fail command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "fail_command",
            skipProperties: true
        });
    });
    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });
    it("should display the help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/fail_help.sh", TEST_ENVIRONMENT);

        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
