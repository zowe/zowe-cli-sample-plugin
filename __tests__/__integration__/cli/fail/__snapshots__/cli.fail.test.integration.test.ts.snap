// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zowe-cli-sample fail command should display the help 1`] = `
"
 DESCRIPTION
 -----------

   [actions] in Zowe CLI are groups of commands. [actions] are always verbs. For
   example, for command \\"zowe zos-files delete data-set\\", the [action] is \\"delete\\".

   For this action (\\"fail\\") we will fail handlers.

 USAGE
 -----

   zowe zowe-cli-sample fail <command>

   Where <command> is one of the following:

 COMMANDS
 --------

   error-handler | eh [object] error-handler always fails!

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-web  | --hw (boolean)

      Display HTML help in browser

"
`;
